curriculum: Change pre and code to blockquote in Advanced Node and Express challenges

thecodingaviator opened the pull request:
<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Refer #35697 #35709

thecodingaviator committed:
curriculum: Change pre and code to blockquote in Advanced Node and Express challenge

thecodingaviator committed:
fix: spacing

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md:
```diff
@@ -9,17 +9,21 @@ challengeType: 2
+&nbsp;&nbsp;if (req.isAuthenticated()) {<br>
+&nbsp;&nbsp;&nbsp;&nbsp;return next();<br>
+&nbsp;&nbsp;}<br>
+&nbsp;&nbsp; res.redirect('/');<br>
```
```suggestion
&nbsp;&nbsp;res.redirect('/');<br>
```

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md:
```diff
@@ -9,17 +9,21 @@ challengeType: 2
-       res.render(process.cwd() + '/views/pug/profile');
-  });</pre>
+<blockquote>
+app.route('/profile')<br>
```
I also debate whether or not to indent the method (chained or not).  Would either of the following be more readable?
```
app.route('/profile').get(
  ensureAuthenticated,
  (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

```
OR not indenting `.get`
```
app.route('/profile')
.get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```

thecodingaviator replied:
I guess not indenting looks better

thecodingaviator replied:
Rather, how doeds this look:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);

RandellDawson replied:
@thecodingaviator Go with the last one above.

thecodingaviator replied:
The one I posted or the latter one you said, I’m sorry a bit confused

RandellDawson replied:
```
app
  .route('/profile')
  .get(ensureAuthenticated, (req,res) => {
    res.render(process.cwd() + '/views/pug/profile');
  }
);
```
Your version.

RandellDawson reviewed (changes requested):
See my suggestions and also my comments below.

For Registration of New Users, should this be two-space indentation (more like below)?
```
app.route('/register')
  .post((req, res, next) => {
    db.collection('users').findOne({ username: req.body.username }, function(err, user) {
      if (err) {
        next(err);
      } else if (user) {
        res.redirect('/');
      } else {
        db.collection('users').insertOne({
          username: req.body.username,
          password: req.body.password
        },
          (err, doc) => {
            if (err) {
              res.redirect('/');
            } else {
              next(null, user);
            }
          }
        )
      }
    })
  },
    passport.authenticate('local', { failureRedirect: '/' }),
    (req, res, next) => {
      res.redirect('/profile');
    }
  );
```

The same question for Implementation of Social Authentication II.
```
passport.use(new GitHubStrategy({
  clientID: process.env.GITHUB_CLIENT_ID,
  clientSecret: process.env.GITHUB_CLIENT_SECRET,
  callbackURL: /*INSERT CALLBACK URL ENTERED INTO GITHUB HERE*/
},
  function(accessToken, refreshToken, profile, cb) {
    console.log(profile);
    //Database logic here with callback containing our user object
  }
));
```

RandellDawson committed:
Remove inadvertent space

Co-Authored-By: thecodingaviator <34807532+thecodingaviator@users.noreply.github.com>

thecodingaviator commented:
About the Registration of new users, I'm fine either way I guess, don't really see how it'll affect anything. What do you think?

RandellDawson commented:
> About the Registration of new users, I'm fine either way I guess, don't really see how it'll affect anything. What do you think?

I like the two space indention that I recommended above for the `Registration of New Users` challenge.

ojeytonwilliams commented:
Sorry to ask this so late in the process, but should we be making them into `<blockquote>`s?  These are code blocks, so the semantic markup is to wrap `<code>` with `<pre>`.

thecodingaviator committed:
Add breaks and nbsp-s

Update registration-of-new-users.english.md

Update implementation-of-social-authentication-ii.english.md

Fix missing breaks

thecodingaviator commented:
@RandellDawson How does it look now?

thecodingaviator commented:
@RandellDawson My bad, how does it look now?

ojeytonwilliams commented:
@RandellDawson Yeah, you're quite right - this wasn't the place to talk about it.  I've created issue #35913 to address it.

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md:
```diff
@@ -10,32 +10,37 @@ As a reminder, this project is being built upon the following starter project on
+&nbsp;&nbsp;&nbsp;&nbsp;(req, res, next) => {<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;res.redirect('/profile');<br>
+&nbsp;&nbsp;&nbsp;&nbsp;}<br>
+&nbsp;&nbsp;);<br>
```
```suggestion
&nbsp;&nbsp;);
```

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/registration-of-new-users.english.md:
```diff
@@ -10,32 +10,37 @@ As a reminder, this project is being built upon the following starter project on
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;res.redirect('/profile');<br>
+&nbsp;&nbsp;&nbsp;&nbsp;}<br>
+&nbsp;&nbsp;);<br>
+);
```
```suggestion
```

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md:
```diff
@@ -9,17 +9,23 @@ challengeType: 2
+&nbsp;.route('/profile')<br>
+&nbsp;.get(ensureAuthenticated, (req,res) => {<br>
+&nbsp;&nbsp;res.render(process.cwd() + '/views/pug/profile');<br>
+&nbsp;}<br>
```
```suggestion
&nbsp;});
```

RandellDawson commented on curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/create-new-middleware.english.md:
```diff
@@ -9,17 +9,23 @@ challengeType: 2
+&nbsp;.get(ensureAuthenticated, (req,res) => {<br>
+&nbsp;&nbsp;res.render(process.cwd() + '/views/pug/profile');<br>
+&nbsp;}<br>
+);
```
```suggestion
```

RandellDawson reviewed (changes requested):
@thecodingaviator I have made two suggestions on two challenges. 

1. Create New Middle - Needed to have the `);` moved up to match the indentation of the `.route` where the `(` originated.

2. Registration of New Users - The suggestion corrects a syntax error that was present in the code.

We these two changes made, we can get this merged.

RandellDawson committed:
Apply suggestions from code review

Co-Authored-By: thecodingaviator <34807532+thecodingaviator@users.noreply.github.com>

RandellDawson commented:
@thecodingaviator Thanks for making those change.  LGTM

ojeytonwilliams commented:
@thecodingaviator Thank you for your contribution to the page! :+1:
We're happy to accept these changes, and look forward to future contributions. :memo:
